Frontend/RecRock-Calendar-Frontend/src/app/calendar/components/calendar/calendar.component.ts

```diff
@@ -105,18 +84,33 @@ export class CalendarComponent {
     return newDate;
   }
 
-  private formatTime(time: Date): string {
-    return time.getHours() + ':' + time.getMinutes();
+  private initialisePopupDTO(
+    userEvent: Event | undefined,
+    date: Date
+  ): PopupDTO {
+    if (userEvent !== undefined) {
+      this.events = this.events.filter((event) => event !== userEvent);
+      return new PopupDTO(
+        this.User.name,
+        PopupDTO.formatTime(userEvent.start),
+        PopupDTO.formatTime(userEvent.end),
+        date,
+        true,
+        false,
+        false,
+        false
+      );
+    } else {
+      return new PopupDTO(
```

Comment:
usually in Angular we use the <Somethig>Model naming instead of DTO, so this can be renamed to PopupModel